tutorials/tmva/TMVA_SOFIE_RDataFrame.py

```diff
@@ -1,62 +1,51 @@
 ### \file
 ### \ingroup tutorial_tmva
 ### \notebook -nodraw
-### This macro provides an example of using a trained model with Keras
-### and make inference using SOFIE and RDataFrame
-### This macro uses as input a Keras model generated with the
-### TMVA_Higgs_Classification.C tutorial
-### You need to run that macro before this one.
-### In this case we are parsing the input file and then run the inference in the same
-### macro making use of the ROOT JITing capability
+### Example of inference with SOFIE and RDataFrame, of a model trained with Keras.
+### First, generate the input model by running `TMVA_Higgs_Classification.C`.
 ###
+### This tutorial parses the input model and runs the inference using ROOT's JITing capability.
 ###
 ### \macro_code
 ### \macro_output
 ### \author Lorenzo Moneta
 
 import ROOT
-
+from os.path import exists
 
 ROOT.TMVA.PyMethodBase.PyInitialize()
 
 
 # check if the input file exists
 modelFile = "Higgs_trained_model.h5"
-if (ROOT.gSystem.AccessPathName(modelFile)) :
-    ROOT.Info("TMVA_SOFIE_RDataFrame","You need to run TMVA_Higgs_Classification.C to generate the Keras trained model")
-    exit()
-
+if (!exists(modelFile)) :
+    raise FileNotFoundError("You need to run TMVA_Higgs_Classification.C to generate the Keras trained model")
 
 # parse the input Keras model into RModel object
 model = ROOT.TMVA.Experimental.SOFIE.PyKeras.Parse(modelFile)
 
-#generatedFile = modelFile
-generatedFile = modelFile.replace(".h5",".hxx")
-print(modelFile, generatedFile)
-#Generating inference code
+# generating inference code
 model.Generate()
-model.OutputGenerated(generatedFile)
+model.OutputGenerated("Higgs_trained_model.hxx")
 model.PrintGenerated()
 
-# now compile using ROOT JIT trained model
-
-inputFileName = "Higgs_data.root"
-inputFile = "http://root.cern.ch/files/" + inputFileName
-
-
+# compile using ROOT JIT trained model
 print("compiling SOFIE model and functor....")
-
+ROOT.gInterpreter.Declare('#include "Higgs_trained_model.hxx"')
 modelName = "Higgs_trained_model"
-ROOT.gInterpreter.Declare('#include "' + generatedFile + '"')
 ROOT.gInterpreter.Declare('auto sofie_functor = TMVA::Experimental::SofieFunctor<7,TMVA_SOFIE_'+modelName+'::Session>(0);')
 
-
+# run inference over input data
+inputFile = "http://root.cern/files/Higgs_data.root"
 df1 = ROOT.RDataFrame("sig_tree", inputFile)
 h1 = df1.Define("DNN_Value", "sofie_functor(rdfslot_,m_jj, m_jjj, m_lv, m_jlv, m_bb, m_wbb, m_wwbb)").Histo1D(("h_sig", "", 100, 0, 1),"DNN_Value")
 
 df2 = ROOT.RDataFrame("bkg_tree", inputFile)
 h2 = df2.Define("DNN_Value", "sofie_functor(rdfslot_,m_jj, m_jjj, m_lv, m_jlv, m_bb, m_wbb, m_wwbb)").Histo1D(("h_bkg", "", 100, 0, 1),"DNN_Value")
 
+# run over the input data once, combining both RDataFrame graphs.
```

Comment:
`ROOT.RDF.RunGraphs....`

Reply:
`RunGraphs` takes a `vector`, so `[...]` it is!